### ISSUE TITLE
How to train other datasets

### ISSUE THREAD

**Issue by aooliver** [EricGuo5513/momask-codes#9]
Hi, thank you for your excellent work. 
I would like to know how to train on other datasets, such as the Motion-X dataset, and what kind of preprocessing operations are required. 
Thank you for your answer.

**Comment by Murrol** [comment]
Hi, thanks for your interest.

In [GET YOUR READ - 3.Get Data](https://github.com/EricGuo5513/momask-codes#round_pushpin-get-you-ready), there are instructions on how to process data. Motion-X also includes the AMASS dataset used in HumanML3D. If you only want to use Motion-X for main-body motion (excluding hands and face), it would be almost the same, except for the difference between SMPL-H and SMPL-X.

For full-body motion, you need to modify the [processing scripts](https://github.com/EricGuo5513/HumanML3D#extract-and-process-data) and the dataloader.

**Comment by LinghaoChan** [comment]
> Hi, thank you for your excellent work. I would like to know how to train on other datasets, such as the Motion-X dataset, and what kind of preprocessing operations are required. Thank you for your answer.

@aooliver Motion-X supports the tomato motion format for model training. Please refer to [this](https://github.com/IDEA-Research/HumanTOMATO/tree/main/src/tomato_represenation).

**Comment by hdjsjyl** [comment]
Hi @LinghaoChan , 
     Are you successful of training momask on motion-x data? I meet the issue "loss nan". Any suggestion from you? Thanks

**Comment by LinghaoChan** [comment]
> Hi @LinghaoChan , Are you successful of training momask on motion-x data? I meet the issue "loss nan". Any suggestion from you? Thanks

@hdjsjyl 

Not yet. However, we have trained HumanTOMATO, MLD, T2M-GPT, MotionDiffuse, and MDM on Motion-X yet. And the TMR was also trained on Motion-X. We plan to release the TMR training/inference on Motion-X soon. You can try the demo [here](https://swanhub.co/demo/Evan/OpenTMR). To make the code easy to use, we are working hard to comment out every potentially questionable code block and encapsulate it as PyPI. Please be patient. Regarding your question, I am not clear about your detailed background now. Sorry about this.  

Ling-Hao CHEN

**Comment by hdjsjyl** [comment]
Hi @LinghaoChan , really appreciate your detailed reply.
    Would you share your training list, val list and testing list?
    In total, I get 81k for whole-body generation, is it correct? Thank you

**Comment by LinghaoChan** [comment]
> Hi @LinghaoChan , really appreciate your detailed reply. Would you share your training list, val list and testing list? In total, I get 81k for whole-body generation, is it correct? Thank you

I remember the training list length is about 64k. Seems good.

**Comment by hdjsjyl** [comment]
hi @LinghaoChan , thanks for your reply. 
    If I want to replace HumanML3D with Motion-X to train the models, such as T2M, MLD. Do you have some suggestions about the training? Should I keep the same training parameters such as learning rate, number of epochs, batch size? Any other concerns I should consider? Thank you

**Comment by LinghaoChan** [comment]
> hi @LinghaoChan , thanks for your reply. If I want to replace HumanML3D with Motion-X to train the models, such as T2M, MLD. Do you have some suggestions about the training? Should I keep the same training parameters such as, kinematic chain, learning rate, number of epochs, batch size? Any other concerns I should consider? Thank you

Seems you are right. The kinematic chain should follow the SMPL-H/-X with tomato format.

**Comment by hdjsjyl** [comment]
hi @LinghaoChan , thank you.
